pkg/kube/pod.go

```diff
@@ -282,14 +282,8 @@ func WaitForPodCompletion(ctx context.Context, cli kubernetes.Interface, namespa
 			return true, err
 		}
 		switch p.Status.Phase {
-		case v1.PodRunning:
-			for _, con := range p.Status.ContainerStatuses {
-				if con.State.Terminated != nil {
-					return false, errors.Errorf("Container %v is terminated, while Pod %v is Running. Container termination status: %v. Pod details (%s)", con.Name, name, con.State.Terminated, p.String())
-				}
-			}
 		case v1.PodFailed:
-			return false, errors.Errorf("Pod %s failed", name)
+			return false, errors.Errorf("Pod %s failed. Pod details (%s)", name, p.String())
```

Comment:
@bathina2  - Does this mean the callers of this function have to ensure that they create their pod using the correct RestartPolicy ? How do we make sure that all consumers switch to the correct policy there?

Reply:
@onkarbhat good catch. I've added the ability to pass down the the restart policy as part of the PodOptions. I've also set the default policy to be Never which was what the original code was intending. Going forward if the consumers want to change the policy they can pass it in via the PodOptions.

Reply:
Discussed this on a call with @tdmanv and @bathina2 . LGTM 👍🏼